### PR TITLE
add goocanvas2 support

### PR DESCRIPTION
**Pull request by zoggy**
opam installation will require installation of conf-goocanvas2 virtual package, already submitted to opam: https://github.com/ocaml/opam-repository/pull/18670

### CODE REVIEW

File: src-goocanvas2/gooCanvas.ml [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+  t
+
+let polyline_line ~x1 ~y1 ~x2 ~y2 ?(props=[]) parent =
+  let t = new polyline (Polyline.new_polyline_line parent#as_item
+    x1 y1 x2 y2)
+  in
+  t#set props ;
+  t
+
+type pixbuf
```
Comment (garrigue) [inline src-goocanvas2/gooCanvas.ml:227]: What is this `pixbuf` type, that doesn't seem to be used anywhere?

File: src-goocanvas2/gooCanvas.ml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+module Tags = struct
+  include (GooCanvasEnums : module type of GooCanvasEnums
+                      with module Conv := GooCanvasEnums.Conv)
+end
+
+let font_desc_conv = (Gobject.Data.unsafe_pointer : Pango.font_description data_conv)
+let canvas_points_conv = (Gobject.Data.unsafe_pointer : GtkGooCanvas.canvas_points data_conv)
+let pointer_events_conv = Gobject.Data.flags GooCanvasEnums.Conv.pointer_events_tbl
+
+let propertize = function
```
Comment (garrigue) [inline src-goocanvas2/gooCanvas.ml:46]: Why is the conversion done by hand rather than use `propcc` ?

**Review by garrigue — COMMENTED** [review]
This looks very useful thanks.
I have just a few questions on the code.

File: src-goocanvas2/gooCanvas.ml (hunk 2 above)
Comment (zoggy) [reply src-goocanvas2/gooCanvas.ml:44]: I took the code of GnoCanvas as example back in 2018 when I started the bindings to goocanvas2, then I did not follow all the changes which occured in lablgtk3. I commited a change to use generated conversion functions, except for canvas-points: I don't know the right way to use propcc to handle this structure https://developer.gnome.org/goocanvas/stable/goocanvas-GooCanvas-Types.html#GooCanvasPoints .

File: src-goocanvas2/gooCanvas.ml (hunk 1 above)
Comment (zoggy) [reply src-goocanvas2/gooCanvas.ml]: Indeed, that's an error. I fix it in the next commit.

File: src-goocanvas2/gooCanvas.ml (hunk 2 above)
Comment (zoggy) [reply src-goocanvas2/gooCanvas.ml:44]: Let me know if there is a more idiomatic way to replace the "propertize" function.

**Pull request merged**